data-prepper-plugins/mapdb-prepper-state/src/main/java/com/amazon/dataprepper/plugins/prepper/state/MapDbPrepperState.java

```diff
@@ -27,20 +27,24 @@ public int compare(byte[] o1, byte[] o2) {
     private static final SignedByteArraySerializer SIGNED_BYTE_ARRAY_SERIALIZER = new SignedByteArraySerializer();
 
     private final BTreeMap<byte[], V> map;
+    private final File dbFile;
 
     public MapDbPrepperState(final File dbPath, final String dbName, final int concurrencyScale) {
+        this.dbFile = new File(String.join("/", dbPath.getPath(), dbName));
         map =
-                (BTreeMap<byte[], V>) DBMaker.fileDB(new File(String.join("/", dbPath.getPath(), dbName)))
+                (BTreeMap<byte[], V>) DBMaker.fileDB(dbFile)
                         .fileDeleteAfterClose()
                         .fileMmapEnable() //MapDB uses the (slower) Random Access Files by default
                         .fileMmapPreclearDisable()
                         .executorEnable()
+                        .transactionEnable()
+                        .closeOnJvmShutdown()
                         .concurrencyScale(concurrencyScale)
                         .make()
                         .treeMap(dbName)
                         .counterEnable() //Treemap doesnt keep:q size counter by default
                         .keySerializer(SIGNED_BYTE_ARRAY_SERIALIZER)
-                        .valueSerializer(Serializer.JAVA).create();
+                        .valueSerializer(Serializer.JAVA).createOrOpen();
```

Comment:
Why do we need this line? Is it because we already created the dbFile on line 33?

Reply:
Yes, this change is to ensure that if we have already created the Db, we open the existing one else it results in an error saying the named db already exists.

Reply:
So nothing about your code makes it so that the db should have already been created right?

Reply:
That's right